sqlglot/dialects/presto.py

```diff
@@ -173,6 +173,29 @@ def _unix_to_time_sql(self: Presto.Generator, expression: exp.UnixToTime) -> str
     return f"FROM_UNIXTIME(CAST({timestamp} AS DOUBLE) / POW(10, {scale}))"
 
 
+def _jsonextract_sql(self: Presto.Generator, expression: exp.JSONExtract) -> str:
+    is_json_extract = self.dialect.settings.get("VARIANT_EXTRACT_IS_JSON_EXTRACT", True)
+
+    if isinstance(is_json_extract, str):
+        # if the kwarg was set by the user it's stored in a string
+        is_json_extract = is_json_extract.lower() == "true"
+
+    # Generate JSON_EXTRACT unless the user has configured that a Snowflake / Databricks
+    # VARIANT extract (e.g. col:x.y) should map to dot notation (i.e ROW access) in Presto/Trino
+    if not expression.args.get("variant_extract") or is_json_extract:
+        return self.func(
+            "JSON_EXTRACT", expression.this, expression.expression, *expression.expressions
+        )
+
+    this = self.sql(expression, "this")
+    expr = self.sql(expression, "expression")
+
+    # Convert the JSONPath extraction `JSON_EXTRACT(col, '$.k1.k2') to a ROW access col.k1.k2
+    expr = expr.replace("'", "").lstrip("$")
```

Comment:
is this `replace` call correct? what if a key has single quotes in it? maybe it would be more robust if we just get what we need from the ast before we generate `expression`

Reply:
I made the assumption here that the `VARIANT` extract notation will not contain special strings, subscripts etc, because in Presto/Trino the dot notation does not seem to handle these.

Would lstripping/rstripping the quotes work better in this case? I'll try out manually generating the path but it will add some code

Reply:
> [...] in Presto/Trino the dot notation does not seem to handle these.

Did you try to use double quotes? Like `x."key with spaces"`.

> Would lstripping + rstripping the quotes work better in this case? I'll try out manually generating the path but it will add some code

I'm not sure why there's a need to strip at all. I see two cases: `JSONExtract.expression` is a string literal, in which case we can simply do `expression.expression.name` to get the path without the surrounding quotes. The other case is that it's not a literal, eg. it's a column, in which case we can't really do something (?).

Reply:
> Did you try to use double quotes? Like x."key with spaces".

Right, this works:

`WITH t AS(SELECT CAST(ROW('value', 1) AS ROW("str1 str2" VARCHAR, num INTEGER)) as col) select col."str1 str2" from t;`

> I'm not sure why there's a need to strip at all.
Yeah, if the JSONPath string is "hidden" behind a column this transformation breaks. I see what you mean with `.name`

Reply:
So how are we going to handle a path stored within a column? What do we currently do?

Reply:
You mean something like the following?

```
with t as (select parse_json('{'z': 1}') as col, 'z' as col2) select col:col2 from t;
```

Such cases seem to be resolved to `NULL` in Snowflake, meaning that `col2` will be resolved as a key and not as `col:"t.col2" -> col:'z'`. 

Even if that were the case, Snowflake does not allow single-quoted identifiers such as `col:'z'` in the VARIANT notation and you can't define an identifier as alias (e.g. `WITH t AS (..., "z" as col2) ...`) to trick it into doing so afaict.

Reply:
Sounds good